desc/compute/_core.py

```diff
@@ -252,33 +256,125 @@ def compute_pressure(
+    Psi : float
+        Total toroidal magnetic flux within the last closed flux surface (Wb).
+    R_transform : Transform
+        Transforms R_lmn coefficients to real space.
+    Z_transform : Transform
+        Transforms Z_lmn coefficients to real space.
+    L_transform : Transform
+        Transforms L_lmn coefficients to real space.
     iota : Profile
         Transforms i_l coefficients to real space.
+    current : Profile
+        Transforms c_l coefficients to real space.
 
     Returns
     -------
     data : dict
         Dictionary of ndarray, shape(num_nodes,) of rotational transform profile.
-        Keys are of the form 'X_y' meaning the derivative of X wrt to y.
+        Keys are of the form 'X_y' meaning the derivative of X wrt y.
 
     """
     if data is None:
         data = {}
 
-    data["iota"] = iota.compute(i_l, dr=0)
-    data["iota_r"] = iota.compute(i_l, dr=1)
-    data["iota_rr"] = iota.compute(i_l, dr=2)
+    if iota is not None:
+        data["iota"] = iota.compute(i_l, dr=0)
+        data["iota_r"] = iota.compute(i_l, dr=1)
+
+    elif current is not None:
+        grid = R_transform.grid
+        data = compute_toroidal_flux(Psi, grid, data=data)
+        data = compute_lambda(L_lmn, L_transform, data=data)
+        data = compute_jacobian(R_lmn, Z_lmn, R_transform, Z_transform, data=data)
+        data = compute_covariant_metric_coefficients(
+            R_lmn, Z_lmn, R_transform, Z_transform, data=data
+        )
+
+        if check_derivs("iota", R_transform, Z_transform, L_transform):
+            # current_term = 2*pi * I / Psi_r = mu_0 / 2*pi * current / psi_r
+            current_term = (
+                mu_0
+                / (2 * jnp.pi)
+                * current.compute(c_l, grid=grid, dr=0)
+                / data["psi_r"]
+            )
+            num = surface_averages(
+                grid,
+                (
+                    data["lambda_z"] * data["g_tt"]
+                    - (1 + data["lambda_t"]) * data["g_tz"]
+                )
+                / data["sqrt(g)"],
+            )
+            den = surface_averages(grid, data["g_tt"] / data["sqrt(g)"])
+            data["iota"] = (current_term + num) / den
+
+        if check_derivs("iota_r", R_transform, Z_transform, L_transform):
+            current_term_r = (
+                mu_0
+                / (2 * jnp.pi)
+                * current.compute(c_l, grid=grid, dr=1)
+                / data["psi_r"]
+                - current_term * data["psi_rr"] / data["psi_r"]
+            )
+            num_r = surface_averages(
+                grid,
+                (
+                    data["lambda_rz"] * data["g_tt"]
+                    + data["lambda_z"] * data["g_tt_r"]
+                    - data["lambda_rt"] * data["g_tz"]
+                    - (1 + data["lambda_t"]) * data["g_tz_r"]
+                )
+                / data["sqrt(g)"]
+                - num * data["sqrt(g)_r"] / data["sqrt(g)"],
```

Comment:
The refactoring is good for `iota` but not for `iota_r`. What we want to do is push the derivative into the surface average. Here we should be using the argument given to the `num` surface average instead of `num`.

We can push in / pull out surface functions (like `psi_r`) from surface average operations. But in general we can't replace a quantity with its surface averaged version inside a surface average. That'll only work for surface averages of quantities that are already surface functions.

Reply:
Thanks for catching that. I was testing with a tokamak example, in which case the `num` term is always 0. We should add tests to automatically catch bugs like this.

Reply:
I’ll add some tests